core/services/log/pool.go

```diff
@@ -29,6 +31,39 @@ func (pool *logPool) addLog(log types.Log) {
 	pool.heap.Insert(Uint64(log.BlockNumber))
 }
 
+func (pool *logPool) getAndDeleteAll() ([]logsOnBlock, int64, int64) {
+	logsToReturn := make([]logsOnBlock, 0)
+	lowest := int64(math.MaxInt64)
+	highest := int64(0)
+
+	for {
+		item := pool.heap.DeleteMin()
+		if item == nil {
+			break
+		}
+
+		blockNum := uint64(item.(Uint64))
+		logs, exists := pool.hashesByBlockNumbers[blockNum]
+		if exists {
+			if int64(blockNum) < lowest {
+				lowest = int64(blockNum)
+			}
+			if int64(blockNum) > highest {
```

Comment:
```suggestion
			} else if int64(blockNum) > highest {
```

Reply:
hm no, it's not ok to have an `else if` - both can be true (usually will) at the same time on the first iteration